frontend/console/src/features/traces/traces.utils.ts

```diff
@@ -0,0 +1,14 @@
+import type { Timestamp } from '@bufbuild/protobuf'
+import type { Event } from '../../protos/xyz/block/ftl/v1/console/console_pb'
+
+export const eventBarLeftOffsetPercentage = (event: Event, requestStartTime: Timestamp | undefined, requestDurationMs: number) => {
+  if (!requestStartTime) {
+    return 0
+  }
+
+  const callTime = event.timeStamp?.toDate() ?? new Date()
+  const initialTime = requestStartTime?.toDate() ?? new Date()
```

Comment:
Is defaulting to the current datetime the behavior we really want here? Should we just return early instead when the input timestamps aren't provided?